lib_eio/net.ml

```diff
@@ -267,3 +269,49 @@ let with_tcp_connect ?(timeout=Time.Timeout.none) ~host ~service t f =
   | exception (Exn.Io _ as ex) ->
     let bt = Printexc.get_raw_backtrace () in
     Exn.reraise_with_context ex bt "connecting to %S:%s" host service
+
+  let accept_fork_domain ~on_error (domain_mgr, domains) (t : #listening_socket) connection_handler () =
+  Switch.run @@ fun sw ->
+  let flow, addr = accept ~sw t in
+  Fun.protect
+    (fun () ->
+      Semaphore.acquire domains ;
+      Domain_manager.run domain_mgr (fun () ->
```

Comment:
I don't understand what this is trying to do. You seem to be spawning a new domain just to handle one connection, which seems highly inefficient.

I would expect `run_server` to start one accept loop for each domain. That's how the existing cohttp-eio version does it, right?

Reply:
Yes, the cohttp-eio has one acceptor per domain.

>You seem to be spawning a new domain just to handle one connection, which seems highly inefficient.

Good point. I completely overlooked the efficiency of spawning domain per connection.

Reply:
the accept loop is similar to cohttp one now.